src/OrbitGl/TimerTrack.cpp

```diff
@@ -103,34 +128,100 @@ void TimerTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick,
+        double end_us = time_graph_->GetUsFromTick(timer_info.end());
+
+        float world_timer_y = GetYFromTimer(timer_info);
+        float box_height = GetTextBoxHeight(timer_info);
+
+        // Check if the previous timer overlaps with the current one, and if so draw the overlap
+        // as triangles rather than as overlapping rectangles.
+        if (k > 0) {
+          const TimerInfo& prev_timer_info = block[k - 1].GetTimerInfo();
+          // We also compare the type, as for the Gpu timers, timers of different type but same
+          // depth are drawn below each other (and thus do not overlap).
+          if (prev_timer_info.end() > timer_info.start() &&
+              prev_timer_info.type() == timer_info.type()) {
+            // Ensure we can distinguish overlapping timers with the same color:
+            if (k % 2 == 0) {
+              color[0] = static_cast<unsigned char>(static_cast<float>(color[0]) * 0.8f);
+              color[1] = static_cast<unsigned char>(static_cast<float>(color[1]) * 0.8f);
+              color[2] = static_cast<unsigned char>(static_cast<float>(color[2]) * 0.8f);
+            }
+
+            double intersection_start_us = start_us;
+            start_us = time_graph_->GetUsFromTick(prev_timer_info.end());
+
+            WorldXInfo world_x_info = ToWorldX(intersection_start_us, start_us, inv_time_window,
+                                               world_start_x, world_width, canvas->GetWidth());
+
+            if (world_x_info.is_visible_width && !is_collapsed) {
+              Vec3 top_left(world_x_info.world_x_start, world_timer_y + box_height, z);
+              float world_intersection_end_x =
+                  world_x_info.world_x_start + world_x_info.world_x_width;
+              Vec3 bottom_right(world_intersection_end_x, world_timer_y, z);
+              Vec3 top_right(world_intersection_end_x, world_timer_y + box_height, z);
+              Triangle triangle(top_left, bottom_right, top_right);
+              auto user_data = std::make_unique<PickingUserData>(
+                  &text_box, [&](PickingId id) { return this->GetBoxTooltip(id); });
+              batcher->AddTriangle(triangle, color, std::move(user_data));
+            }
+          }
+        }
+
+        // Check if the next timer overlaps with the current one, and if so draw the overlap
+        // as triangles rather than as overlapping rectangles.
+        if (k + 1 < block.size()) {
+          const TimerInfo& next_timer_info = block[k + 1].GetTimerInfo();
+          // We also compare the type, as for the Gpu timers, timers of different type but same
+          // depth are drawn below each other (and thus do not overlap).
+          if (timer_info.end() > next_timer_info.start() &&
+              next_timer_info.type() == timer_info.type()) {
+            double intersection_end_us = end_us;
+            end_us = time_graph_->GetUsFromTick(next_timer_info.start());
```

Comment:
I found a little bit confusing the intersection_end_us and the end_us variables.

Could end_us be renamed as intersection_start_us??

Reply:
Added a new variable for this.

Reply:
Should be better now.